Cython/Compiler/Options.py

```diff
@@ -644,6 +646,10 @@ def __init__(self, defaults=None, **kw):
             options['language_level'] = directives['language_level']
         elif not options.get('language_level'):
             options['language_level'] = directive_defaults.get('language_level')
+        if 'legacy_implicit_noexcept' in directives and 'legacy_implicit_noexcept' not in kw:
+            options['legacy_implicit_noexcept'] = directives['legacy_implicit_noexcept']
+        elif not options.get('legacy_implicit_noexcept'):
```

Comment:
Doesn't this also hit the case where `legacy_implicit_noexcept=False`? I mean, it's certainly ok to overwrite it with a `False` default directives value in that case, but then we're assuming here that both default values actually are False. The code here probably shouldn't depend on that. Maybe we could use `None` for the default value instead of `False`, and test that here?

Reply:
Just to note - a lot of this goes away if we keep it out of `options` and just extract it from `compiler_directives` in `Context.__init__`

Reply:
Yeah, that sounds reasonable.

Reply:
Implemented.